src/Mealz/MealBundle/Service/ParticipationServiceTrait.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Mealz\MealBundle\Service;
+
+use App\Mealz\MealBundle\Entity\Meal;
+use App\Mealz\MealBundle\Entity\Participant;
+use App\Mealz\MealBundle\Entity\ParticipantRepository;
+use App\Mealz\MealBundle\Entity\Slot;
+use App\Mealz\MealBundle\Entity\SlotRepository;
+use DateTime;
+
+/**
+ * @property ParticipantRepository $participantRepo
+ * @property SlotRepository        $slotRepo
+ */
+trait ParticipationServiceTrait
+{
+    private function getNextFreeSlot(DateTime $mealDate): ?Slot
+    {
+        $slots = $this->slotRepo->findBy(['disabled' => 0, 'deleted' => 0], ['order' => 'ASC']);
+        if (1 > count($slots)) {
+            return null; // no active slots are available; return null
```

Comment:
The first part of the comment is fine. The "return null" is obvious, I think, we don't need to repeat this.